code/controllers/subsystem/lighting.dm

```diff
@@ -32,18 +32,22 @@ SUBSYSTEM_DEF(lighting)
 
 	var/list/queue = sources_queue
 	var/i = 0
-	for (i in 1 to length(queue))
-		var/datum/light_source/L = queue[i]
+	while(i < length(queue)) //we don't use for loop here because i cannot be changed during an iteration
+		i += 1
```

Comment:
Back to basics :)

Reply:
Ok so about the C style for loop, it's a problem
![Screenshot (182)](https://user-images.githubusercontent.com/110812394/235344672-1e9a04ff-05c2-4257-8a1b-f21d47e2735d.png)
This happens because in the for loop `i = 1` first get's executed before checking `i <= length(queue)` so even if the list is empty i always get's set to 1 and if condition get's falsely executed at all times causing the out of bounds

Reply:
god fucking damn it byond